### DISCUSSION TITLE
How to split different UpstreamHttpMethod and same UpstreamPathTemplate ?

### DISCUSSION THREAD

**Discussion by hsmadi** [ThreeMammals/Ocelot#1866] [Q&A] [answered]
I'm trying to manage permissions for the same upstream url based on ClaimRoute, how can I achieve the following:
```json
{
      "DownstreamPathTemplate": "/User/{username}",
      "DownstreamScheme": "https",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 9901
        }
      ],
      "UpstreamPathTemplate": "/api/user/{username}",
      "UpstreamHttpMethod": [ "Get", "Post", "Put" ],
      "AuthenticationOptions": {
        "AuthenticationProviderKey": "Bearer",
        "AllowedScopes": []
      }
    },
```
Second user scenario:
```json
    {
      "DownstreamPathTemplate": "/User/{username}",
      "DownstreamScheme": "https",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 9901
        }
      ],
      "UpstreamPathTemplate": "/api/user/{username}",
      "UpstreamHttpMethod": [ "Delete", "Patch" ],
      "AuthenticationOptions": {
        "AuthenticationProviderKey": "Bearer",
        "AllowedScopes": []
      },
      "RouteClaimsRequirement": {
        "Role": "Administrator"
      }
    }
```
I couldn't find anywhere in the documentation where we can split RouteClaims based on method.

Is that possible in the first place, and is there a way to deal with this using Ocelot, because I believe someone had this issue before.

**Comment by raman-m** [comment, marked as answer]
Hi @hsmadi !
You posted unexpected question! 🤣 

> Is that possible in the first place, and is there a way to deal with this using Ocelot, because I believe someone had this issue before.

`"UpstreamPathTemplate": "/api/user/{username}",`
So, different route behaviors means different routes!
**You cannot merge all user scenarios into one route with one upstream template!**

To solve that problem allowing different behaviors, you have to define different upstream templates.
For the 2nd scenario, Why not to specify another template like this: `"UpstreamPathTemplate": "/api/user/{username}/auth",` ?
That means this path requires the Admin role and given HTTP methods.

Hope it helps!
